pkg/streamingpromql/types/ring_buffer_test.go

```diff
@@ -286,6 +286,51 @@ func TestRingBuffer_RemoveLastPoint(t *testing.T) {
 	})
 }
 
+func TestRingBuffer_ViewUntilWithExistingView(t *testing.T) {
+	t.Run("FPoint ring buffer", func(t *testing.T) {
+		buf := NewFPointRingBuffer(limiting.NewMemoryConsumptionTracker(0, nil))
+		require.NoError(t, buf.Append(promql.FPoint{T: 1, F: 100}))
+		require.NoError(t, buf.Append(promql.FPoint{T: 2, F: 200}))
+		require.NoError(t, buf.Append(promql.FPoint{T: 3, F: 300}))
+		require.NoError(t, buf.Append(promql.FPoint{T: 4, F: 400}))
+
+		view := buf.ViewUntilSearchingForwards(2, nil)
+		viewShouldHavePoints(t, view, promql.FPoint{T: 1, F: 100}, promql.FPoint{T: 2, F: 200})
+
+		// Test that reusing the view with ViewUntilSearchingForwards works correctly.
+		view = buf.ViewUntilSearchingForwards(1, view)
```

Comment:
should we ignore the returned value here to confirm that `view` is being reused and not a new value is being returned into view (either that or checking the pointer values etc)

Reply:
Yep, fixed in af9b9fa